### PR TITLE
OP-19592 - Added JUNIT Test cases for ISD Trail SAAS user registration

### PR DESCRIPTION
**Pull request by Pranav-b-7**
https://devopsmx.atlassian.net/browse/OP-19592

### CODE REVIEW

File: opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        responseModel.setEventProcessed(true);
+        responseModel.setEventId(eventId);
+
+        ReflectionTestUtils.setField(accountSetupService, "spinnakerAutomationWebhookURL", "http://localhost:8084/webhook/start/spinnaker");
+        ReflectionTestUtils.setField(accountSetupService, "argoAutomationWebhookURL", "http://localhost:8084/webhook/start/argo");
+    }
+
+    @Test
+    @DisplayName("Should store the user details in DB for CD type isdArgo")
+    void test1(){
```
Comment (rahul-chekuri) [inline opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java:63]: we are not asserting anything here?

File: opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+        Mockito.when(userRepository.findByBusinessEmailAndCdType(datasourceRequestModel.getBusinessEmail(), CDType.isdArgo)).thenReturn(Optional.empty());
+        accountSetupService.store(datasourceRequestModel, CDType.isdArgo);
+    }
+
+    @Test
+    @DisplayName("Should store the user details in DB for CD type isdSpinnaker")
+    void test2(){
+
```
Comment (rahul-chekuri) [inline opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java:72]: same here

File: opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        Mockito.when(userRepository.findByBusinessEmailAndCdType(datasourceRequestModel.getBusinessEmail(), CDType.isdSpinnaker)).thenReturn(Optional.empty());
+        accountSetupService.store(datasourceRequestModel, CDType.isdSpinnaker);
+    }
+
+    @Test
+    @DisplayName("Should not fail even if the given user details is already present in the DB for cd type isdArgo")
+    void test3(){
+
```
Comment (rahul-chekuri) [inline opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java:80]: same here

File: opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        users.add(user);
+
+        Mockito.when(userRepository.findByBusinessEmailAndCdType(datasourceRequestModel.getBusinessEmail(), CDType.isdArgo)).thenReturn(Optional.of(users));
+        accountSetupService.store(datasourceRequestModel, CDType.isdArgo);
+    }
+
+    @Test
+    @DisplayName("Should not fail even if the given user details is already present in the DB for cd type isdSpinnaker")
+    void test4(){
+
```
Comment (rahul-chekuri) [inline opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java:100]: same here

File: opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java (hunk 1 above)
Comment (Pranav-b-7) [reply opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java:63]: This is a test case for void method and hence there are no values returned to assert anything here. However i have added a verify which are commonly used for void methods.

File: opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java (hunk 2 above)
Comment (Pranav-b-7) [reply opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java:73]: This is a test case for void method and hence there are no values returned to assert anything here. However i have added a verify which are commonly used for void methods.

File: opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java (hunk 3 above)
Comment (Pranav-b-7) [reply opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java:82]: This is a test case for void method and hence there are no values returned to assert anything here. However i have added a verify which are commonly used for void methods.

File: opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java (hunk 4 above)
Comment (Pranav-b-7) [reply opsmx-isd-register/src/test/java/com/opsmx/isd/register/service/AccountSetupServiceImplTest.java:103]: This is a test case for void method and hence there are no values returned to assert anything here. However i have added a verify which are commonly used for void methods.

**Review by rahul-chekuri — APPROVED** [review]

**Pull request merged**